flow/energy_models/base_energy.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABCMeta, abstractmethod
+
+class BaseEnergyModel(metaclass=ABCMeta):
```

Comment:
I don't think you need to have this ABCMeta thing. I'm a little rusty on metaclasses and might be wrong though, so first, my question for you is -- why did you put this here?

Reply:
ah this was my suggestion. i made a bunch of the base classes throughout the codebase into abstract base classes (that was landed in master, not sure if it trickled down to i210_dev). basically i wanted any instantiation of a base class to error right away, rather than erroring when an un-implemented method within is called.